server/storage/players.lua

```diff
@@ -382,8 +418,97 @@ local function cleanPlayerGroups()
     lib.print.info('Removed invalid groups from player_groups table')
 end
 
+---@param citizenid string
+---@param key string | string[]
+---@param value any
+local function addPlayerDataUpdate(citizenid, key, value)
+    local hasSubKeys = type(key) == 'table'
+
+    if hasSubKeys then
+        key[1] = otherNamedPlayerFields[key[1]] or key[1]
+    else
+        key = otherNamedPlayerFields[key] or key
+    end
+
+    if jsonPlayerFields[hasSubKeys and key[1] or key] == nil then
+        lib.print.error(('Tried to update player data field %s when it doesn\'t exist. Value: %s'):format(hasSubKeys and key[1] or key, value))
+        return
+    end
+
+    if hasSubKeys and not jsonPlayerFields[key[1]] then
+        lib.print.error(('Tried to update player data field %s as a json object when it isn\'t one'):format(key[1]))
+        return
+    end
+
+    value = type(value) == 'table' and json.encode(value) or value
+
+    -- In sendPlayerDataUpdates we don't go more than 3 tables deep
+    if hasSubKeys and #key > 4 then
+        lib.print.error(('Cannot save field %s because data is too big.\nkeys: %s\nvalue: %s'):format(key[1], json.encode(key), value))
+        return
+    end
+
+    local currentTable = isUpdating and playerDataUpdateQueue or collectedPlayerData
+    if not currentTable[citizenid] then
+        currentTable[citizenid] = {}
+    end
+
+    currentTable[citizenid][hasSubKeys and key[1] or key] = hasSubKeys and {} or value
+
+    if hasSubKeys then
+        local current = currentTable[citizenid][key[1]]
+        if #key > 2 then
+            -- We don't check the last one because otherwise we lose the table reference
+            for i = 2, #key - 1 do
+                if not current[key[i]] then
+                    current[key[i]] = {}
+                end
```

Comment:
```suggestion
                    current[key[i]] = {}
                    current = current[key[i]]
                end
```
Also this, we need to update the `current` so we can index the last nested key and insert the value